plutus-tx-plugin/src/PlutusTx/Compiler/Kind.hs

```diff
@@ -16,6 +16,10 @@ import qualified PlutusCore              as PLC
 compileKind :: Compiling uni fun m => GHC.Kind -> m (PLC.Kind ())
 compileKind k = withContextM 2 (sdToTxt $ "Compiling kind:" GHC.<+> GHC.ppr k) $ case k of
     -- this is a bit weird because GHC uses 'Type' to represent kinds, so '* -> *' is a 'TyFun'
-    (GHC.isLiftedTypeKind -> True)        -> pure $ PLC.Type ()
-    (GHC.splitFunTy_maybe -> Just (i, o)) -> PLC.KindArrow () <$> compileKind i <*> compileKind o
-    _                                     -> throwSd UnsupportedError $ "Kind:" GHC.<+> GHC.ppr k
+    (GHC.isLiftedTypeKind -> True)         -> pure $ PLC.Type ()
+    (GHC.splitFunTy_maybe -> Just (i, o))  -> PLC.KindArrow () <$> compileKind i <*> compileKind o
+    -- We ignore pi types to handle '(#, #)' kind
```

Comment:
The problem with this sort of comment is that it says that the code is _related_ to handling unboxed tuples, but it doesn't say enough for a reader to understand _why_ this code helps with unboxed tuple support. Which is why we need a note, so we can write a slightly lengthier explanation and refer to it repeatedly.